sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -191,50 +190,544 @@ def get_next(next_link=None):
+                raise HttpResponseError(response=response, model=error)
+
+            return pipeline_response
+
+        return ItemPaged(get_next, extract_data)
+
+    @overload
+    def update_manifest_properties(self, repository, tag_or_digest, properties, **kwargs):
+        # type: (str, str, ArtifactManifestProperties, **Any) -> ArtifactManifestProperties
+        pass
+
+    @overload
+    def update_manifest_properties(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> ArtifactManifestProperties
+        pass
+
     @distributed_trace
-    def get_artifact(self, repository_name, tag_or_digest, **kwargs):
-        # type: (str, str, Dict[str, Any]) -> RegistryArtifact
-        """Get a Registry Artifact object
+    def update_manifest_properties(self, *args, **kwargs):
+        # type: (Union[str, ArtifactManifestProperties], **Any) -> ArtifactManifestProperties
+        """Set the properties for a manifest
 
-        :param str repository_name: Name of the repository
-        :param str tag_or_digest: The tag or digest of the artifact
-        :returns: :class:`~azure.containerregistry.RegistryArtifact`
-        :raises: None
+        :param args:
+        :type args: Union[str, ~azure.containerregistry.ArtifactManifestProperties]
+        :param str repository: Repository the manifest belongs to
+        :param str tag_or_digest: Tag or digest of the manifest
+        :param properties: The property's values to be set
+        :type properties: :class:`~azure.containerregistry.ArtifactManifestProperties`
+        :returns: :class:`~azure.containerregistry.ArtifactManifestProperties`
+        :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
+
+        Example
+
+        .. code-block:: python
+            from azure.containerregistry import ContainerRepositoryClient
+            from azure.identity import DefaultAzureCredential
+            account_url = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+            client = ContainerRepositoryClient(account_url, "my_repository", DefaultAzureCredential())
+            for artifact in client.list_manifests():
+                received_properties = client.update_manifest_properties(
+                    artifact.digest,
+                    ManifestWriteableProperties(
+                        can_delete=False,
+                        can_list=False,
+                        can_read=False,
+                        can_write=False,
+                    ),
+                )
+        """
+        repository = args[0]
+        tag_or_digest = args[1]
+        properties = None
+        if len(args) == 3:
+            properties = args[2]
+        else:
+            properties = ArtifactManifestProperties(
+                can_delete=kwargs.pop("can_delete", None),
+                can_list=kwargs.pop("can_list", None),
+                can_read=kwargs.pop("can_read", None),
+                can_write=kwargs.pop("can_write", None),
+            )
```

Comment:
We should combine these:
```python
properties = None
if len(args) == 3:
    properties = args[2]
else:
    properties = ArtifactManifestProperties()
properties.can_delete = kwargs.pop("can_delete", properties.can_delete)
...  
```
Otherwise if someone provides both the model and the kwarg they wont be popped and will fail at the transport.